src/domain/repository/marketplaceQueryRepo.go

```diff
@@ -1,20 +1,21 @@
 package repository
 
 import (
+	"github.com/goinfinite/os/src/domain/dto"
 	"github.com/goinfinite/os/src/domain/entity"
-	"github.com/goinfinite/os/src/domain/valueObject"
 )
 
 type MarketplaceQueryRepo interface {
-	ReadCatalogItems() ([]entity.MarketplaceCatalogItem, error)
-	ReadCatalogItemById(
-		catalogId valueObject.MarketplaceItemId,
+	ReadCatalogItems(
+		requestDto dto.ReadMarketplaceCatalogItemsRequest,
+	) (dto.ReadMarketplaceCatalogItemsResponse, error)
+	ReadOneCatalogItem(
+		requestDto dto.ReadMarketplaceCatalogItemsRequest,
 	) (entity.MarketplaceCatalogItem, error)
-	ReadCatalogItemBySlug(
-		slug valueObject.MarketplaceItemSlug,
-	) (entity.MarketplaceCatalogItem, error)
-	ReadInstalledItems() ([]entity.MarketplaceInstalledItem, error)
-	ReadInstalledItemById(
-		installedId valueObject.MarketplaceItemId,
+	ReadInstalledItems(
+		requestDto dto.ReadMarketplaceInstalledItemsRequest,
+	) (dto.ReadMarketplaceInstalledItemsResponse, error)
+	ReadOneInstalledItem(
+		requestDto dto.ReadMarketplaceInstalledItemsRequest,
```

Comment:
On interfaces, you only need to name the variables if you have multiple and the name is the differentiating or explicating factor. In this case, "requestDto" present or not, the result is the same, therefore it's redundant.